flytekit/clients/grpc_utils/wrap_exception_interceptor.py

```diff
@@ -0,0 +1,48 @@
+import typing
+from typing import Union
+
+import grpc
+
+from flytekit.exceptions.base import FlyteException
+from flytekit.exceptions.system import FlyteSystemException
+from flytekit.exceptions.user import (
+    FlyteAuthenticationException,
+    FlyteEntityAlreadyExistsException,
+    FlyteEntityNotExistException,
+    FlyteInvalidInputException,
+)
+
+
+class RetryExceptionWrapperInterceptor(grpc.UnaryUnaryClientInterceptor, grpc.UnaryStreamClientInterceptor):
+    def __init__(self, max_retries: int = 3):
+        self._max_retries = 3
+
+    @staticmethod
+    def _raise_if_exc(request: typing.Any, e: Union[grpc.Call, grpc.Future]):
+        if e.code() == grpc.StatusCode.UNAUTHENTICATED:
+            raise FlyteAuthenticationException() from e
```

Comment:
Am I missing something, where is this handled?

Reply:
this is only an exception beautifier. This is not handled anywhere, this is once all auth tries fail